test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -33,12 +33,72 @@ Describe "Split Operator" -Tags CI {
             $res[2] | Should -Be "c"
             $res[3] | Should -Be "d"
 
+            $res = "a b c d" -split " ", -2
+            $res.count | Should -Be 2
+            $res[0] | Should -Be "a b c"
+            $res[1] | Should -Be "d"
+
             $res = "a b c d" -split " ", -1
+            $res.count | Should -Be 1
+            $res[0] | Should -Be "a b c d"
+        }
+
+        It "Binary split operator can work with different delimeter than split string" {
+            $res = "a b c d" -split " ",8
             $res.count | Should -Be 4
             $res[0] | Should -Be "a"
             $res[1] | Should -Be "b"
             $res[2] | Should -Be "c"
             $res[3] | Should -Be "d"
+
+            $res = "a b c d" -split " ",-8
```

Comment:
Why not reverse order like test in line 80?

Reply:
Agreed this is something that I noticed today while showing off my changes to the judges. I will probably come back and add another commit fixing this functionality.

Which way do you think is better? Personally I think that the first item should be at array index 0.
ie
```
> $r = 'a b c d' -split ' ',-2
> $r[0]
d
```

Reply:
Any formatting or implementation details would be greatly appreciated. Thank you! :)

Reply:
Note that splitting from right to left might be considered orthogonal to the iteration from left to right.

@mklement0's original example in #4765 is:

```
PS> 'a b c d' -split ' ', -2  # split into (at most) 2 strings from the end
a b c  # prefix
d        # requested token

PS> 'a b c d' -split ' ', -3  # split into (at most) 3 strings from the end
a b   # prefix
c       
d 

PS> 'a b' -split ' ', -2  # 2 resulting strings - complete split; same as 0 in this case
a
b
```

This implementation currently accords with that.

Reply:
Also worth noting that suggestion was approved by the @PowerShell/powershell-committee

Reply:
Test in line 80 is wrong, yes?

Reply:
@iSazonov Yes you were correct it is fixed in the most recent commits.
@rjmholt  Good point, I don't have a good response for that haha.